tests/internal/test_serverless.py

```diff
@@ -78,7 +79,40 @@ def test_not_azure_function_consumption_plan():
     assert in_azure_function() is False
 
 
-def test_slow_imports():
+standard_blocklist = [
+    "ddtrace.appsec._api_security.api_manager",
+    "ddtrace.appsec._iast._ast.ast_patching",
+    "ddtrace.internal.telemetry.telemetry_writer",
+    "email.mime.application",
+    "email.mime.multipart",
+    "logging.handlers",
+    "multiprocessing",
+    "importlib_metadata",
+    # These modules must not be imported because their source files are
+    # specifically removed from the serverless python layer.
+    # See https://github.com/DataDog/datadog-lambda-python/blob/main/Dockerfile
+    "ddtrace.appsec._iast._taint_tracking._native",
+    "ddtrace.appsec._iast._stacktrace",
+    "ddtrace.internal.datadog.profiling.libdd_wrapper",
+    "ddtrace.internal.datadog.profiling.ddup._ddup",
+    "ddtrace.internal.datadog.profiling.stack_v2._stack_v2",
+]
+expanded_blocklist = standard_blocklist + [
+    "importlib.metadata",
+]
+
+
+@pytest.mark.parametrize(
+    "package,blocklist",
+    [
+        ("ddtrace", expanded_blocklist),
+        ("ddtrace.contrib.internal.aws_lambda", expanded_blocklist),
+        ("ddtrace.contrib.internal.psycopg", expanded_blocklist),
+        # requests imports urlib3 which imports importlib.metadata
+        ("ddtrace.contrib.requests", standard_blocklist),
```

Comment:
@brettlangdon I'm wondering if it would be possible to run this test importing all the packages in `ddtrace.contrib`. Right now, none of these import paths are tested and I'm concerned that our build scripts might be removing important binaries which these require. Right now, we'd have no way of knowing if say `import ddtrace.contrib.flask` attempts to import one of the removed binaries listed above.

Reply:
🤔 I am not sure to be honest, because just importing `ddtrace.contrib.requests` might not actually import anything. Since most of the `ddtrace.contrib.<integration>.__init__` files will check if the necessary packages are installed first before trying to load the integration patching code. So it might basically be a no-op and then not actually test what you need.


e.g. you might need to actually install all of these packages first to then test them.


if we consider this, we might want to move the serverless tests out of `tests/internal` and re-think how they are structured/run. There are some `appsec_iast_packages` tests (`tests/appsec/iast_packages`) which basically install a ton of packages to test against (not sure we should copy/pasta that code, but example to consider).